File: Main_project/Client/src/main/webapp/scripts/constants/country.js
/**
 * Created by vlad on 08.05.15.
 */


var Country = {
	UKRAINE: new CountryConstructor(1, "Украина"),
	RUSSIA: new CountryConstructor(2, "Россия"),
	GEORGIA: new CountryConstructor(3, "Грузия"),
	values: function () {
		var values = [];
		for (var item in Country) {
			if (typeof Country[item] == "object") {
				values[values.length] = Country[item];
			}
		}
		return values;
	},
	getById: function (id) {
		var values = Country.values();
		for (var i = 0; i < values.length; i++) {
			var item = values[i];
			if (item.getId() == id) {
				return item;
			}
		}
	}
};

function CountryConstructor(id, name) {
	this.getId = function () {
		return id;
	};
	this.getName = function () {
		return name;
	};
}

File: Main_project/Server/Containers/src/main/java/com/pb/dashboard/Country.java
package com.pb.dashboard;

public enum Country {

    UKRAINE(1, "Украина"),
    RUSSIA(2, "Россия"),
    GEORGIA(3, "Грузия");

    private int id;
    private String name;

    private Country(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public static Country pkeyToCountry(String idStr) {
        int pkey = Integer.parseInt(idStr);
        return pkeyToCountry(pkey);
    }

    public static Country pkeyToCountry(int id) {
        for (Country country : Country.values()) {
            if (country.getId() == id) {
                return country;
            }
        }
        throw new IllegalArgumentException("Country with id [" + id + "] not exists.");
    }
}